From b57ab61417c3480c9887e2cc67d3c4bdd2750cdb Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 10 Aug 2023 17:26:01 +0200
Subject: [PATCH 337/841] refactor(stm32mp2): configure clocks and UART earlier
 in BL31

DT, clocks and console init are done earlier in the function
bl31_early_platform_setup2().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 86eaddfee8f65624c98f76485ff9273446af543c refactor(stm32mp2): configure clocks and UART earlier in BL31]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I3aa8c440e842134551211998d5ca43597d420ebb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/327285
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 plat/st/stm32mp2/bl31_plat_setup.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index fdde782d1..1b99ae38f 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -45,6 +45,16 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 
 	configure_mmu();
 
+	if (dt_open_and_check(arg1) < 0) {
+		panic();
+	}
+
+	if (stm32mp2_clk_init() < 0) {
+		panic();
+	}
+
+	(void)stm32mp_uart_console_setup();
+
 	assert(params_from_bl2 != NULL);
 	assert(params_from_bl2->h.type == PARAM_BL_PARAMS);
 	assert(params_from_bl2->h.version >= VERSION_2);
@@ -80,16 +90,6 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 		bl_params = bl_params->next_params_info;
 	}
 
-	if (dt_open_and_check(arg1) < 0) {
-		panic();
-	}
-
-	if (stm32mp2_clk_init() < 0) {
-		panic();
-	}
-
-	(void)stm32mp_uart_console_setup();
-
 	/*
 	 * This stm32mp_get_soc_name() call is mandatory here to store SoC part number,
 	 * and avoid using BL31 DT at runtime to check stm32mp_is_single_core().
-- 
2.47.2

